src/main/java/com/connexta/store/adaptors/StorageAdaptor.java

```diff
@@ -39,4 +39,12 @@ void store(
    */
   @NotNull
   StorageAdaptorRetrieveResponse retrieve(@NotBlank final String key) throws RetrieveException;
+
+  /**
+   * Updates the {@link StoreStatus} of {@code dataset} for a given {@code datasetId}
+   *
+   * @param datasetId the Id of a dataset
+   * @param storeStatus the new {@link StoreStatus} for a dataset
+   */
+  void updateStatus(String datasetId, String storeStatus);
```

Comment:
❗️ What happens if the datasetId does not exist? What happens if there is an error reading the file? Client needs to be able to respond to these cases.

Reply:
I'll add a check for the datasetId. The error reading the file would be an exception internal to S3. What's weird is that the javadoc for setObjectTagging doesn't say it throws anything. https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3.html#setObjectTagging-com.amazonaws.services.s3.model.SetObjectTaggingRequest-